deepchecks/vision/vision_data.py

```diff
@@ -255,9 +278,11 @@ def copy(self, n_samples: int = None, shuffle: bool = False, random_state: int =
                                                                   shuffle=shuffle,
                                                                   random_state=random_state,
                                                                   n_samples=n_samples)
-        # If new data is sampled, then needs to remove cached info
+        # If new data is sampled, then needs to re-calculate cache
         if n_samples:
-            new_vision_data._n_of_samples_per_class = None
+            new_vision_data.init_cache()
+            for batch in new_vision_data:
+                new_vision_data.update_cache(self.batch_to_labels(batch))
```

Comment:
When does this happen? because we use copy to create the sampled data obj?

Reply:
yes